internal/resource/demo/demo.go

```diff
@@ -84,6 +85,32 @@ func Register(r resource.Registry) {
 		}
 	}
 
+	validateArtistFn := func(res *pbresource.Resource) error {
+		if proto.Equal(res.Id.Type, TypeV1Artist) {
+			artist := &pbdemov1.Artist{}
+			if err := anypb.UnmarshalTo(res.Data, artist, proto.UnmarshalOptions{}); err != nil {
+				return err
+			}
+			if artist.Name == "" {
+				return fmt.Errorf("artist.name required")
```

Comment:
Out of scope for this PR, but seeing this makes me wonder whether we should have some basic helpers for constructing validation errors.

For example:

```Go
return resource.RequiredFieldError("name")
```